backend/canisters/user_index/impl/src/updates/register_user.rs

```diff
@@ -87,39 +81,20 @@ fn prepare(args: &Args, runtime_state: &mut RuntimeState) -> Result<PrepareOk, R
         _ => {}
     };
 
-    let cycles_to_use = if runtime_state.data.canister_pool.is_empty() {
-        let cycles_required = USER_CANISTER_INITIAL_CYCLES_BALANCE + CREATE_CANISTER_CYCLES_FEE;
-        if !utils::cycles::can_spend_cycles(cycles_required, MIN_CYCLES_BALANCE) {
-            return Err(CyclesBalanceTooLow);
-        }
-        cycles_required
-    } else {
-        0
-    };
-
     if !runtime_state.data.users.reserve_username(&args.username) {
         return Err(UsernameTaken);
     }
 
-    let canister_id = runtime_state.data.canister_pool.pop();
-    let canister_wasm = runtime_state.data.user_canister_wasm.clone();
-    let init_canister_args = InitUserCanisterArgs {
-        owner: caller,
-        group_index_canister_id: runtime_state.data.group_index_canister_id,
-        notifications_canister_ids: runtime_state.data.notifications_canister_ids.clone(),
-        ledger_canister_id: runtime_state.data.ledger_canister_id,
-        wasm_version: canister_wasm.version,
-        username: args.username.clone(),
-        test_mode: runtime_state.data.test_mode,
-    };
-
-    Ok(PrepareOk {
-        caller,
-        canister_id,
-        canister_wasm,
-        cycles_to_use,
-        init_canister_args,
-    })
+    if let Some(local_user_index_canister) = runtime_state.data.local_index_map.index_for_new_user() {
+        let user_wasm_version = runtime_state.data.user_canister_wasm.version;
+        Ok(PrepareOk {
+            local_user_index_canister,
+            user_wasm_version,
```

Comment:
Does the user index need to know the version of each user? This may not be the actual version that gets installed due to it being async.

Reply:
Only to show in the UI